scene/3d/visual_instance_3d.cpp

```diff
@@ -41,7 +43,38 @@ void VisualInstance3D::_update_visibility() {
 		return;
 	}
 
-	RS::get_singleton()->instance_set_visible(get_instance(), is_visible_in_tree());
+	bool already_visible = _is_vi_visible();
+	bool visible = is_visible_in_tree();
+	_set_vi_visible(visible);
+
+	// If making visible, make sure the rendering server is up to date with the transform.
+	if (visible && !already_visible) {
+		if (!_is_using_identity_transform()) {
+			Transform3D gt = get_global_transform();
+			RS::get_singleton()->instance_set_transform(instance, gt);
+		}
```

Comment:
```suggestion
	if (visible && !already_visible && !_is_using_identity_transform()) {
		Transform3D gt = get_global_transform();
		RS::get_singleton()->instance_set_transform(instance, gt);
```
IMO long if conditions are easier to read than deeply nested blocks

Reply:
I prefer it the way it is. I implemented all your other style suggestions, thanks!